### ISSUE TITLE
L1/L2 Regularisation

### ISSUE THREAD

**Issue by jackaraz** [danielward27/flowjax#197]
Hi, how can I add L1/L2 regularisation to the loss functions? I couldn't manage to extract the parameters to compute the regularisation. It would be really helpful if you could provide a small example.

Thanks

**Comment by danielward27** [comment]
That would not be expected, at least for me, the below only prints ``float32``.

```python
import jax.numpy as jnp
import equinox as eqx
from flowjax.flows import masked_autoregressive_flow
from flowjax.distributions import Normal
import jax.random as jr
import jax

flow = masked_autoregressive_flow(
    jr.key(0),
    base_dist=Normal(jnp.zeros(3)),
)
params, static = eqx.partition(flow, eqx.is_inexact_array)
leaves = jax.tree_util.tree_leaves(params)

[print(leaf.dtype) for leaf in leaves]
```
Did you use  ``eqx.is_array`` instead of ``eqx.is_inexact_array``?  If that isn't the issue and you can create a minimal example I can have a look.

**Comment by jackaraz** [comment]
ah, no, my bad. Thanks for the explanation that clarifies it :)
